src/RootContainer.js

```diff
@@ -1,21 +1,81 @@
 import React, { useState, useEffect } from 'react';
 import { queryData } from './query';
 import GeneOntologyNetwork from './components/GeneOntologyNetwork';
+import Controls from './components/Controls';
+import Loading from './components/Loading';
 
 const RootContainer = ({ serviceUrl }) => {
 	const [data, setData] = useState([]);
+	const [ontologyList, setOntologyList] = useState([]);
+	const [selectedOntology, changeOntology] = useState('');
+	const [selectedOntologyData, setOntologyData] = useState({});
+	const [loading, setLoading] = useState(true);
+
 	useEffect(() => {
+		setLoading(true);
 		queryData({
 			serviceUrl: serviceUrl,
 			geneId: '1205472,128,2314'
 		}).then(data => {
 			setData(data);
+			setLoading(false);
 		});
 	}, []);
 
+	useEffect(() => {
+		const list = [];
+		data.forEach(d =>
+			d.goAnnotation.forEach(g => {
+				list.push(g.ontologyTerm.namespace);
+			})
+		);
+		setOntologyList([...new Set(list)]);
+	}, [data]);
+
+	useEffect(() => {
+		const filteredMap = ontologyList.reduce(
+			(curMap, ontology) => (
+				(curMap[ontology] = data.map(item => ({
```

Comment:
Can you change the reduce function to also use the spread operator? This is more common, and mutating the accumulator can cause surprises in some instances.
```
			(curMap, ontology) => ({
				...curMap,
				[ontology]: data.map(,,,)
			}),
```

Reply:
yes, what you suggested is better as -
1. Object mutation is a bad thing to do, better create new instances.
2. The new code looks more cleaner and concise.